File: 2928/2928.go
func distributeCandiesAmongChildren(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := distributeCandiesAmongChildren(root.Left)
	r := distributeCandiesAmongChildren(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 2928/2928.cs
public class Solution {
    public int DistributeCandiesAmongChildren(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(DistributeCandiesAmongChildren(root.left), DistributeCandiesAmongChildren(root.right));
    }
}